src/firmware/app/control/trajectory_planner.h

```diff
@@ -261,11 +284,21 @@ void static app_trajectory_planner_generateTimeProfile(
  *
  *  TODO: Remove when #1322 is merged.
  *
- * @pre forwards is pre-allocated up to at least TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS
- * @param forwards This is the trajectory that will be modified in place to become a
- * reverse trajectory
- * @param num_segments The number of segements(elements) in the forwards array
+ * @param forwards_trajectory This is the trajectory that will be modified in place to
+ * become a reverse trajectory
  */
 void static app_trajectory_planner_reverseTrajectoryDirection(
-    TrajectoryElement_t forwards[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
-    const unsigned int num_segments);
+    PositionTrajectory_t* forwards_trajectory);
+
+/**
+ * This function generates a velocity trajectory that corresponds to the time-optimal
+ * velocity to follow a specified path. This profile is based on the input position
+ * trajectory
+ *
+ * @param position_trajectory [in] The position trajectory to build the velocity
+ * trajectory from
+ * @param velocity_trajectory [out] The velocity trajectory that corresponds to the
+ * time-optimal velocity to follow a specified path
+ */
+void app_trajectory_planner_generateVelocityTrajectory(
```

Comment:
If VelocityTrajectory is in the name, then the other functions should have Position in the name for consistency imo

Reply:
:+1:

Renamed the functions I thought this would be relevant to